```python
# 최초의 solution. test case도 통과 못하고 runtime error
def compare(prev_num, num):
    return int(prev_num + str(num)) < int(str(num) + prev_num)


def solution(numbers):
    res, success = [], False
    numbers = sorted(numbers, reverse=True)
    for number in numbers:
        if res == []:
            res.append(str(number)); continue
        for ix, prev_num in enumerate(res):
            if compare(prev_num, number):
                res.insert(ix, str(number))
                success = True
                break
        if not success:
            res.append(str(number))
        success = False
    return ''.join(res)
```
